### ISSUE TITLE
EPG missing on (private) channels

### ISSUE THREAD

**Issue by herrnst** [nirvana-7777/pvr.magenta#8]
As **galitus** already mentioned in the UHD channels issue in https://github.com/nirvana-7777/pvr.magenta/issues/7#issue-2352269912, I'm seeing the "guide for the private channels is not loading" issue aswell (though for quite some time now, unfortunately never got to create an issue report for this):

On most (if not all) "Öffentlich-rechtliche", all EPG data is in place just as it should, but most (if not all) "private" channels (RTL, Pro7, ~~also the Fussball.TV 1-3 channels~~) are missing all EPG info completely, means the guide is completely empty for this.

Not sure what exactly is required in terms of debug logs, or a copy of the actual Epgxx.db files from Kodi, so please advise on that. What I can tell is that I'm using pvr.magenta in parallel with my Tvheadend server and pvr.plutotv, which both display proper EPG data in Kodi. I have the "prefer HD channels" option disabled due to the Linux DRM level issues, and the start channel is set to 3001 (so 3001 = Das Erste HD). Kodi is configured to pick and use channel numbers as reported by the backend, with the "use channel numbers from backend in a multi-client setup" option enabled aswell. And: This is using Magenta 2.0.

On a side note, scrolling through the guide window gets extremely slow when pvr.magenta channels are visible, but this is probably something for another issue report.

EDIT: Using the 21.9.5 code from today with the UHD "fix".

Correction: Fussball.TV 1 now displays EPG, and while testing on the laptop, I also noticed EPG data on Fussball.TV 2 and 3.

**Comment by herrnst** [comment]
The issue of missing EPG data very likely relates to the "prefer HD channels" option (which I - as a plain Linux user - have disabled due to the known fact of the DRM levels on most private channels):

I just made a test by disabling pvr.magenta, then additionally delete all EPG data from within the Kodi GUI. Then enabled the "prefer HD channels" option and activated pvr.magenta again. All channels were populated with EPG data.

The same test but vice versa regarding the "prefer HD channels" option (deactivate addon, clear, disable option, activate), and the channels affected by that option were without EPG data.
